doc/beamline_completion.md

```diff
@@ -71,25 +71,84 @@ This is a breaking change so would have to happen at some controlled point.
+| 3(c)  | 1 | 3 | y | Solution similar to the above, but fields are optional |
+| 4 |  12 | 32 | n | Almost total removal as set out in above plan |
+| 5 | 6 | 14 | n |Complete removal and clean-up |
+| **Total Weeks** | 24 | 64 | | | 
+
+## Dataset Objectives
+
+The `Dataset` objectives are slightly different from everything outlined above. Firstly `Dataset` provides a more "green field" approach rather than the "brown field" that has to be considered when supporting existing workspaces. For example `Dataset` rollout will involve a new stream of non-compatible (with existing base `API::Algorithm`) `Algorithms` and that gives significant flexibility when breaking with old behaviour. 
+
+Here are some key differences as relating to the Geometry:
+
+* Fundamental non-compatibility between `Workspace` and `Dataset`. `Dataset` has more information than possible in `Workspace`. All interop via conversion, API compatibility not maintained.
+  - Existing saving would have to "slice" `Dataset`, so would be required to write new saving functionality and format for `Dataset`
+  - Because of saving requirement, no requirement to save geometry deltas. Geometry can be written to be directly correspond to the mutated Beamline geometry - not origninal (IDF) + delta
+  - Loading functionality would follow from the above
+* No ExperimentInfo
+  - Geometry, alongside run information is part of `Dataset`
+* No ParameterMap
+  - Exact solution(s) not yet apparent, but there are several options. Given that a number of parameters are linked to the instrument and not individual components, this might affect how things are done
+* No `ComponentID` by design. Infact, there may be no requirement for unique identification of `Components` at all.
+
+Another key difference with the phases for non-dataset approach is that the approach concludes with the complete removal of `Instrument 1.0` but not `Workspace 1.0`. The current preferred rollout for `Dataset` is to develop alongside the existing codebase, and not to consider removal of legacy data structures i.e. Workspace until `Dataset` usage has acted as a full replacement. We would simultaneously remove `Workspace 1.0`, `Algorithms 1.0`, `Instrument 1.0`, `ExperimentInfo`. 
+
+The net result of these differences is that the activities and timescales differ from those presented in the above section. We refer to these as activities rather than phases, since the order of the activities is not known and possibly not interdependent.
+
+### Activity 1 - Functional wrappers to Dataset for Geometry
+
+Need to provide classes of the form `SpectrumInfo`, `ComponentInfo` and `DetectorInfo` that wrap `Dataset`. These classes should point to a `Dataset` containing all required information. Note that `ComponentInfo` and `DetectorInfo` are inter-dependent, a higher-level grouping may be required. `SpectrumInfo` uses `DetectorInfo`. We would need to consider time spent to reproduce functional and performance testing.
```

Comment:
@SimonHeybrock Good point. Are there considerations for the `DataSetView` representing say a Detector if the bank is moved via `ComponentInfo`?

Reply:
Like other iterator-like objects, `DatasetView` would get invalidated by calling a non-const method of the dataset. Since `DatasetView` is a lightweight object created on-the-fly this should not be an issue --- the view should not be long-lived anyway.